packages/vite/src/node/plugins/assetImportMetaUrl.ts

```diff
@@ -56,15 +57,28 @@ export function assetImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
             const ast = this.parse(rawUrl)
             const templateLiteral = (ast as any).body[0].expression
             if (templateLiteral.expressions.length) {
-              const pattern = JSON.stringify(buildGlobPattern(templateLiteral))
+              const pattern = buildGlobPattern(templateLiteral)
+              if (pattern.startsWith('**')) {
+                config.logger.warnOnce(
+                  colors.yellow(
+                    `${colors.cyan(
+                      `new URL(${rawUrl}, import.meta.url)`
+                    )} matches most files in system. It will be left as-is.`
+                  )
+                )
+                continue
+              }
```

Comment:
Just checked the issue, and this fix works for me 👍 I think it's fine without a warning as a start since it's rare someone would use template literals like this.

Reply:
I merged main branch and removed the warning 👍